userguide/tutorials/plugin_development.adoc

```diff
@@ -497,4 +531,8 @@ mvn clean install -Dcheck.fail-enforcer=false -Dcheck.fail-dependency=false -Dch
 
 However, this is *not recommended*, we recommend that you fix the POM file and run the build with all the checks in place.
 
+=== java.lang.NoClassDefFoundError or java.lang.ClassNotFoundException
+
+Sometimes, when you develop a custom plugin, a `java.lang.NoClassDefFoundError` or a `java.lang.ClassNotFoundException` exception may occur on starting Kill Bill. This is most probably because the class in question is not present on the classpath. 
 
+For a plugin to work, any classes used by the plugin must be present on the classpath. So, the class needs to be present in the *plugin jar* itself or it must be *imported* from Kill Bill. Refer to the <<Importing Additional Packages in Plugins>> section above for further details.
```

Comment:
Going back to my remark above, for customers using a different JVM, they may need to also export specific packages from Kill Bill using `org.killbill.osgi.system.bundle.export.packages.extra`.